src/models/xml_shape_parser.cpp

```diff
@@ -35,18 +36,18 @@ geo::ShapePtr parseXMLShape(const std::string& filename, std::string& error)
 {
     std::stringstream s_error;
 
-    TiXmlDocument doc(filename);
-    doc.LoadFile();
+    tinyxml2::XMLDocument doc;
+    doc.LoadFile(filename.c_str());
 
     if (doc.Error())
     {
         s_error << "While parsing '" << filename << "': " << std::endl << std::endl
-                << doc.ErrorDesc() << " at line " << doc.ErrorRow() << ", col " << doc.ErrorCol() << std::endl;
+                << doc.ErrorStr() << " at line " << doc.ErrorLineNum() << std::endl;
         error = s_error.str();
         return geo::ShapePtr();
     }
 
-    const TiXmlElement* model_xml = doc.FirstChildElement("model");
+    const tinyxml2::XMLElement* model_xml = doc.FirstChildElement("model");
```

Comment:
Could we used `unique_ptr`?

Reply:
You could probably get it working. But I don't think it is faster. I prefer to keep it this way for now.